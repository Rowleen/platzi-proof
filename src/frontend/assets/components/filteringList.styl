@import "styles/variables"

.genres-wrapper
  display: flex
  margin: 3rem 0
  justify-content: space-between
  align-items: center

  .genres
    display: flex
    flex: 1
    flex-flow: row
    padding-left: 2rem
    align-items: center

    .pill-clear
      cursor: pointer
      align-items: center
      height: 1.8rem
      border-radius: border-radius-elements
      padding: .1rem .8rem
      margin-right: .4rem
      text-align: center
      color: white
      background-color: rgba(0, 0, 0, .5)
      transition: all .2s ease
      display: none

      &.show
        display: flex

      &:hover
        background-color:rgba(0, 0, 0, .65)
        border-color: primary-color

        >.icon-times
          color: white
          transform: rotate(270deg)

      .icon-times
        color: primary-color
        margin-left: .5rem
        transition: transform .25s ease-in

  .sorter-wrapper
    display: flex
    justify-content: flex-end
    align-items: center

    .sort-button
      cursor: pointer
      display: flex
      align-items: center
      background-color: rgba(0, 0, 0, .3)
      border-radius: border-radius-elements
      color: white
      border: unset
      padding: .4rem 1rem
      z-index -1
      opacity 0

      &:hover
        background-color: rgba(0, 0, 0, .5)
        color: secondary-color

      &:focus
        outline: none

      &.show-sort
        z-index 0
        opacity 1

      .icon-sort
        margin-left: .5rem

        &.desc
          transform: rotate(180deg)