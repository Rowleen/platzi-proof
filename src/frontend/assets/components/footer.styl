@import 'styles/variables'

.footer
  width: 100%
  display: flex
  justify-content: center
  background-color: background-color-accent
  padding: 1rem 1rem

  .footer-content
    width: 100%
    max-width: 1200px
    color: text-color
    font-size: 1rem