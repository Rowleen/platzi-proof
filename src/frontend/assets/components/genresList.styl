@import "styles/variables"

.genres-list
  display: flex
  flex-flow: row wrap
  list-style: none

  .genre
    cursor: pointer
    height: 1.8rem
    display: flex
    align-items: center
    border-radius: border-radius-elements
    padding: .1rem .8rem
    margin-right: .4rem
    text-align: center
    color: white
    border: unset

    &:nth-child(1n+0)
      background-color: third-color
      &:hover
        background-color: third-color-hover

    &:nth-child(2n+0)
      background-color: secondary-color
      &:hover
        background-color: secondary-color-hover

    &:nth-child(3n+0)
      background-color: primary-color
      &:hover
        background-color: primary-color-hover

@media (max-width: 414px)
  .genres-list .genre
    margin-bottom: .4rem
