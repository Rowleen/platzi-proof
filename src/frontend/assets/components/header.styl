@import 'styles/variables'
  
.header
  width: 100vw
  height: auto
  display: flex;
  justify-content: center
  background-color: white
  padding: 0.5rem 1rem
  box-shadow: 0 0.25rem 0.25rem rgba(0, 0, 0, 0.2)

  .header-content
    width: 100%
    max-width: 1200px
    height: auto
    display: grid
    grid-template-columns: repeat(auto-fill, minmax(50%, 1fr))

    .site-title
      color: $dark-grey