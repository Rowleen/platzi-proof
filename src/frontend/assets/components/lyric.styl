@import "styles/variables"

.lyric-wrapper
  position: fixed
  top: 0
  left: 0
  width: 100vw
  height: 100vh
  display: flex
  justify-content: center
  align-items: center
  opacity: 0
  z-index: -1
  transition: all .2s ease
  background-color: rgba(255, 255, 255, .2)

  &.show-lyric
    opacity: 1
    z-index: 1

  .lyric-container
    position: relative
    width: 100%
    max-width: 40rem
    height: auto
    background-color: background-color-accent
    border-radius: border-radius-containers
    
    .close-lyric
      cursor: pointer
      position: absolute
      top: 1rem
      right: 1rem
      color: white
      font-size: 2rem
    
    .lyric-body
      height: auto
      color: white
      overflow-y: auto 
      padding: 2rem

