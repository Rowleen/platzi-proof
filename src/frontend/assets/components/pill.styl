@import "styles/variables"

.pill
  cursor: pointer
  height: 1.8rem
  border-radius: border-radius-elements
  padding: .1rem .8rem
  margin-right: .4rem
  text-align: center
  color: white
  margin-bottom: .4rem

  &:nth-child(1n+0)
    background-color: third-color
    &:hover
      background-color: third-color-hover

  &:nth-child(2n+0)
    background-color: secondary-color
    &:hover
      background-color: secondary-color-hover

  &:nth-child(3n+0)
    background-color: primary-color
    &:hover
      background-color: primary-color-hover