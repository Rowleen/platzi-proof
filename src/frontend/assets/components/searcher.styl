@import "styles/variables"

.form
  width: 100%
  height: auto
  display: flex;
  justify-content: center
  align-items: center
  flex-direction: column

  &.complete
    height calc(100vh - 11rem)

  &:focus-within
    .form-elements
      animation: pulse-disco 2s cubic-bezier(0.26, 1.36, 0.74, -0.29) infinite;
      border: solid 1px
      border-color: primary-color

      .input, .select, .button .icon
        color white

  .form-elements
    display: flex
    border: solid 1px background-color-accent
    width: 35rem
    border-radius: border-radius-elements
    box-shadow 0 0.25rem 0.25rem rgba(backgroundColor)

    .input
      width: 23rem
      height: 3rem
      border: solid 1px lightGrey
      padding: .5rem 0 .5rem 1.5rem
      background-color: background-color-accent;
      border-bottom-left-radius: border-radius-elements
      border-top-left-radius: border-radius-elements
      font-size: 1.5rem
      color: white
      border-color: background-color-accent
      border-right: unset
      outline: unset
      transition: all .2s ease

      &:focus
        &::placeholder
          color: white

    .select
      width: 8rem
      font-size: 1rem
      color: white
      padding: 0 .8rem
      color: disabled-text
      border: solid 1px
      border-left: unset
      border-right: unset
      outline: unset
      border-color: background-color-accent;
      background-color: background-color-accent;
      transition: all .2s ease

      &:focus
        border: solid 1px white

    .button
      cursor: pointer
      width: 4rem
      display: flex
      align-items: center
      justify-content: center
      background-color: background-color-accent
      border: solid 1px
      border-left: unset
      border-color: background-color-accent
      border-top-right-radius: border-radius-elements
      border-bottom-right-radius: border-radius-elements
      transition: all .2s ease
      outline: unset

      &:focus
        border: solid 1px white

      &:hover
        background-color: rgba(background-color-accent, .4)

      .icon
        font-size: 1.5rem
        color: disabled-text

  .welcome-text
    width: 35rem
    color: white
    margin-top: 1rem
    text-align: center

    &.hide-welcome
      display: none

    &.info-text
      color: secondary-color

@keyframes pulse-disco
  0%
    box-shadow 0 0 0.25rem rgba(primary-color, 1)
  25%
    box-shadow 0 .25rem .25rem rgba(secondary-color, 1)
  50%
    box-shadow 0 0 0.25rem rgba(primary-color-hover, 1)
  75%
    box-shadow 0 .25rem 0.25rem rgba(third-color, 1)
  100%
    box-shadow 0 0 0.25rem rgba(secondary-color-hover, 1)