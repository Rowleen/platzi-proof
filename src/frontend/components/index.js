export { default as FilteringList } from "./FilteringList";
export { default as Footer } from "./Layout/Footer";
export { default as GenresList } from "./GenresList";
export { default as Header } from "./Layout/Header";
export { default as Layout } from "./Layout/Layout";
export { default as Lyric } from "./Lyric";
export { default as Pill } from "./Pill";
export { default as Spinner } from "./Spinner";
export { default as Track } from "./Track";
export { default as TrackList } from "./TrackList";
