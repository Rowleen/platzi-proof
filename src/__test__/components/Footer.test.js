/* eslint-disable no-undef */
import React from "react";
import { create } from "react-test-renderer";
import { mount } from "enzyme";

import { Footer } from "components";

describe("Testing <Footer />", () => {
  const footer = mount(<Footer />);

  test("Render of the component", () => {
    expect(footer.length).toEqual(1);
  });

  test("Render of the text", () => {
    expect(footer.find(".footer-content").text()).toEqual(
      "Developed with ðŸ’š for Platzi"
    );
  });

  test("Check if the UI not change", () => {
    const footerUi = create(<Footer />);
    expect(footerUi.toJSON()).toMatchSnapshot();
  });
});
