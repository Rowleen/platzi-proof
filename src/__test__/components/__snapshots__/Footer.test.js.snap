// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Footer /> Check if the UI not change 1`] = `
<footer
  className="footer"
>
  <div
    className="footer-content"
  >
    Developed with ðŸ’š for Platzi
  </div>
</footer>
`;
